### DISCUSSION TITLE
Can't we play (starttime, endtime) in this way now?

### DISCUSSION THREAD

**Discussion by lvhaoyu39** [katspaugh/wavesurfer.js#3150] [Q&A]
How to set the starttime and endtime for playback now.

**Comment by katspaugh** [comment]
You can use setTime to set the start time and the timeupdate event to check if it reached the end time.

**Reply by lvhaoyu39** [reply]
I set the start time and timeupdate to monitor whether the end time has been reached, but there is a time error between them that cannot be accurately determined and the playback is paused.

**Reply by lvhaoyu39** [reply]
Can I refer to any relevant examples?

**Reply by katspaugh** [reply]
You should listen to the `finish` event for that. See https://wavesurfer.xyz/docs/types/wavesurfer.WaveSurferEvents

**Reply by lvhaoyu39** [reply]
This finish event is used to monitor the end of audio playback. The difference between this and the end of audio playback is that the audio in a certain interval of the entire audio is played randomly. Currently, it is not possible to accurately monitor whether the end time has been reached.

**Reply by katspaugh** [reply]
Ah, I see. You want to schedule playback from time A to time B.
It used to be possible to accurately schedule playback with Web Audio, but now we rely on HTML Audio which isn't very accurate when it comes to tracking time. 
I might restore the option to use Web Audio, I already have a prototype here: https://github.com/katspaugh/wavesurfer-multitrack/blob/main/src/webaudio.ts

**Reply by lvhaoyu39** [reply]
Yes, currently I have only used a relatively clumsy method of comparing time to one decimal place and then pausing, but there are slight differences in the waveform.
